common_code/inspiral/CE_energy.f90

```diff
@@ -425,12 +434,23 @@ subroutine CE_inject_case5(id, ierr)
                F_drag = ((beta-0.9)*F_drag_supersonic + (0.99-beta)*F_drag_subsonic)/0.09
             endif
 
+            ! Add hydrodynamical drag
+            F_drag = F_drag + pi * (CE_companion_radius * Rsun)**2 * rho_at_companion * v_rel**2
+
             mdot = mdot_HL * 10.0**log_mdot_factor
             s% xtra(23) = mdot
 
             ! Accretion luminosity luminosity: 10% efficiency
             L_acc = 0.1 * standard_cgrav * M2 / R2 * mdot_HL
+         
+         else if (s% x_integer_ctrl(2) == 4) then
 
+            if (s% x_logical_ctrl(7)) then 
+               ! Accretion luminosity luminosity: 10% efficiency
+               L_acc = (s% xtra(30) * Msun * Rsun / (secyer**2.0) )/(s% xtra(16) * Rsun / secyer)
```

Comment:
Could you please use variable names instead of the xtra array? This makes the code (changes) easier to understand.